src/components/GoalDependencyForm/GoalDependencyForm.tsx

```diff
@@ -41,8 +41,9 @@ interface GoalDependencyAddFormProps {
 }
 
 export const GoalDependencyAddForm: React.FC<GoalDependencyAddFormProps> = ({ goalId, kind, isEmpty, onSubmit }) => {
-    const [selected, setSelected] = useState<GoalByIdReturnType | null>(null);
     const [query, setQuery] = useState('');
+    const selected = useRef<GoalByIdReturnType | null>(null);
```

Comment:
It looks tricky and it's a bad practice to pass ref as prop (<GoalSuggest ... showSuggest={!selected.current} />) because `ref` updating doesn't trigger re-render. Why do you need `selected` here? You can get selected goal id from [`getValues()`](https://react-hook-form.com/docs/useform/getvalues)